src/fluree/db/graphdb.cljc

```diff
@@ -64,60 +64,64 @@
-                     (conj post f)
-                     post)
-                   (if (get ref?-map (.-p f))
-                     (conj opst f)
-                     opst)
-                   ecount*)))))))
+  (go
+    (try*
+      (let [t                    (.-t ^Flake (first flakes))
+            _                    (when (not= t (dec (:t db)))
+                                   (throw (ex-info (str "Invalid with called for db " (:dbid db) " because current 't', " (:t db) " is not beyond supplied transaction t: " t ".")
+                                                   {:status 500
+                                                    :error  :db/unexpected-error})))
+            add-flakes           (filter #(not (exclude-predicates (.-p ^Flake %))) flakes)
+            add-preds            (into #{} (map #(.-p ^Flake %) add-flakes))
+            idx?-map             (into {} (map (fn [p] [p (dbproto/-p-prop db :idx? p)]) add-preds))
+            ref?-map             (into {} (map (fn [p] [p (dbproto/-p-prop db :ref? p)]) add-preds))
+            flakes-bytes         (flake/size-bytes add-flakes)
+            schema-change?       (schema-util/schema-change? add-flakes)
+            root-setting-change? (schema-util/setting-change? add-flakes)
+            pred-ecount          (-> db :ecount (get const/$_predicate))
+            add-pred-to-idx?     (if schema-change? (schema-util/add-to-post-preds? add-flakes pred-ecount) [])
+            db*                  (loop [[add-pred & r] add-pred-to-idx?
+                                        db db]
+                                   (if add-pred
+                                     (recur r (<? (add-predicate-to-idx db add-pred)))
+                                     db))
+            ;; this could require reindexing, so we handle remove predicates later
+            db*                  (-> db*
+                                     (assoc :t t)
+                                     (update-in [:stats :size] + flakes-bytes) ;; total db ~size
+                                     (update-in [:stats :flakes] + (count add-flakes)))]
+        (loop [[^Flake f & r] add-flakes
+               spot   (get-in db* [:novelty :spot])
+               psot   (get-in db* [:novelty :psot])
+               post   (get-in db* [:novelty :post])
+               opst   (get-in db* [:novelty :opst])
+               ecount (:ecount db)]
+          (if-not f
+            (let [db*  (assoc db* :ecount ecount
+                                  :novelty {:spot spot :psot psot :post post :opst opst
+                                            :size (+ flakes-bytes (get-in db* [:novelty :size]))})
+                  db** (if (or schema-change? (nil? (:schema db*)))
+                         (assoc db* :schema (<? (schema/schema-map db*)))
+                         db*)]
+              (cond-> db**
+
+                      root-setting-change?
+                      (assoc :settings (<? (schema/setting-map db**)))))
```

Comment:
minor suggestion, but i think the following might be a little simpler than this `let` block:

```clojure
(cond-> (-> db*
            (assoc-in [:novelty :size] :spot spot, :psot psot, :post post, :opst opst)
            (update-in [:novelty :size] + flakes-bytes)
            (assoc :ecount ecount))
  (or schema-change?
      (nil? (:schema db*))) (assoc :schema (<? (schema/schema-map db*)))

  root-setting-change?      (assoc :settings (<? (schema/setting-map db**))))
```

how much simpler this suggestion is over what's here is debatable though, so take it or leave it.